api/async.rb

```diff
@@ -103,4 +107,40 @@ def validate
       end
     end
   end
+
+  # Many resources are async but do not provide a long running Operation that can be
+  # polled against. This class is meant to poll the resource itself until it can be
+  # determined that the resource is in a ready state.
+  class PollAsync < Async
```

Comment:
nit: Since Operations also involve polling, wdyt of `ResourceStateAsync` or `StateAsync`? That was they're name {{Thing being polled}}Async.

Reply:
I'm not sure that's better. Resource and State are also shared verbage for an Async Operation. IMO an async resource that requires polling is distinct to supporting operations. Since I can see the argument either way I'll let @rambleraptor be the tie break.

Reply:
I'm going to go with @chrisst on this one.

Reply:
sgtm